### PR TITLE
Upgrade parent, plugin dependencies

### PR DESCRIPTION
**Pull request by sghill**
Hi, I see this plugin has been deprecated recently. We have quite a few jobs that rely on it, so I was curious what it would take to bring it up to the latest plugin version vs converting to pipelines. It ended up being straightforward, so I thought I'd submit this PR to see if there was interest in merging.

The bulk of the work was in getting compliant with the spotbugs rules that come enabled by default in parent version 4.40. These are mainly around preventing NPEs, exception handling, and stream closing.
I chose to use Jenkins 2.319.3 as a baseline based on [the guide](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/).
I bumped several plugins that were not included in the plugins-bom based on [security advisories](https://www.jenkins.io/security/advisories/).

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Closes #227 
Closes #237 
Closes #238 
Closes #241 
Closes #242 
Closes #247 
Closes #246

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -251,8 +286,7 @@
 
 	<properties>
 		<findbugs.failOnError>false</findbugs.failOnError>
-				<workflow.version>1.6</workflow.version>
-                <jenkins.version>2.120</jenkins.version>
+                <jenkins.version>2.319.3</jenkins.version>
 		<java.level>8</java.level>
```
Comment (NotMyFault) [inline pom.xml:290]:
```suggestion
```
This is now obsolete within plugin 4.40

File: pom.xml [hunk 2]
```diff
@@ -251,8 +286,7 @@
 
 	<properties>
 		<findbugs.failOnError>false</findbugs.failOnError>
-				<workflow.version>1.6</workflow.version>
-                <jenkins.version>2.120</jenkins.version>
+                <jenkins.version>2.319.3</jenkins.version>
```
Comment (NotMyFault) [inline pom.xml:289]:
```suggestion
                <jenkins.version>2.319.1</jenkins.version>
```
There's been a lot of back and forth about point releases of LTS lines, but I'd prefer to use .1 for now.

**Comment by NotMyFault** [comment]
Hey,

thanks for the PR, I labeled this plugin as deprecated, but some modernization can't hurt, if people want to continue using it.
I went ahead and lifted the restriction from the Jenkinsfile, to test this PR on ci.jenkins.io too.

**Comment by sghill** [comment]
Thanks for the review and additional improvements @NotMyFault! Really appreciate you taking the time to share the context here on `java.level` and LTS patch versions so I can help make the next one smoother.

**Review by NotMyFault — APPROVED** [review]
Thanks a bunch for your contribution!

**Pull request merged**